common/types/encoding/da.go

```diff
@@ -0,0 +1,232 @@
+			Nonce:    txData.Nonce,
+			To:       txData.To,
+			Value:    txData.Value.ToInt(),
+			Gas:      txData.Gas,
+			GasPrice: txData.GasPrice.ToInt(),
+			Data:     data,
+			V:        txData.V.ToInt(),
+			R:        txData.R.ToInt(),
+			S:        txData.S.ToInt(),
+		})
+
+	case types.AccessListTxType:
+		tx = types.NewTx(&types.AccessListTx{
+			ChainID:    txData.ChainId.ToInt(),
+			Nonce:      txData.Nonce,
+			To:         txData.To,
+			Value:      txData.Value.ToInt(),
+			Gas:        txData.Gas,
+			GasPrice:   txData.GasPrice.ToInt(),
+			Data:       data,
+			AccessList: txData.AccessList,
+			V:          txData.V.ToInt(),
+			R:          txData.R.ToInt(),
+			S:          txData.S.ToInt(),
+		})
+
+	case types.DynamicFeeTxType:
+		tx = types.NewTx(&types.DynamicFeeTx{
+			ChainID:    txData.ChainId.ToInt(),
+			Nonce:      txData.Nonce,
+			To:         txData.To,
+			Value:      txData.Value.ToInt(),
+			Gas:        txData.Gas,
+			GasTipCap:  txData.GasTipCap.ToInt(),
+			GasFeeCap:  txData.GasFeeCap.ToInt(),
+			Data:       data,
+			AccessList: txData.AccessList,
+			V:          txData.V.ToInt(),
+			R:          txData.R.ToInt(),
+			S:          txData.S.ToInt(),
+		})
+
+	case types.L1MessageTxType:
+	default:
+		return nil, fmt.Errorf("unsupported tx type: %v", txData.Type)
+	}
+
+	rlpTxData, err := tx.MarshalBinary()
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal binary of the tx: %+v, err: %w", tx, err)
+	}
+
+	return rlpTxData, nil
+}
+
+// GetCrcMax calculates the maximum row consumption of crc.
+func (c *Chunk) GetCrcMax() uint64 {
+	// Map sub-circuit name to row count
+	crc := make(map[string]uint64)
+
+	// Function to accumulate row consumption
+	add := func(rowConsumption *types.RowConsumption) {
+		if rowConsumption == nil {
+			return
+		}
+		for _, subCircuit := range *rowConsumption {
+			crc[subCircuit.Name] += subCircuit.RowNumber
+		}
+	}
+
+	// Function to find maximum row consumption
+	max := func() uint64 {
+		var maxVal uint64
+		for _, value := range crc {
+			if value > maxVal {
+				maxVal = value
+			}
+		}
+		return maxVal
+	}
+
+	// Iterate over blocks, accumulate row consumption
+	for _, block := range c.Blocks {
+		if block.RowConsumption != nil {
+			add(block.RowConsumption)
+		}
+	}
+
+	// Return the maximum row consumption
+	return max()
+}
+
+// GetNumTransactions calculates the total number of transactions in a Chunk.
+func (c *Chunk) GetNumTransactions() uint64 {
+	var totalTxNum uint64
+	for _, block := range c.Blocks {
+		totalTxNum += uint64(len(block.Transactions))
+	}
+	return totalTxNum
+}
+
+// GetNumL2Transactions calculates the total number of L2 transactions in a Chunk.
+func (c *Chunk) GetNumL2Transactions() uint64 {
+	var totalTxNum uint64
+	for _, block := range c.Blocks {
+		totalTxNum += block.NumL2Transactions()
+	}
+	return totalTxNum
+}
+
+// GetNumL2GasUsed calculates the total gas of L2 transactions in a Chunk.
+func (c *Chunk) GetNumL2GasUsed() uint64 {
```

Comment:
```suggestion
func (c *Chunk) GetL2GasUsed() uint64 {
```

Reply:
changed in [3337b05](https://github.com/scroll-tech/scroll/pull/1178/commits/3337b055f920fb9e445f9f713e42e42ea648850b)